crates/server/src/verifysnsignature.rs

```diff
@@ -0,0 +1,313 @@
+use std::str::FromStr;
+use std::vec;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use starknet_core::types::{BlockId, BlockTag, Felt, FunctionCall};
+use starknet_core::utils::get_selector_from_name;
+use starknet_crypto::{poseidon_hash_many, verify};
+use starknet_providers::jsonrpc::HttpTransport;
+use starknet_providers::{JsonRpcClient, Provider, Url};
+
+// Structure representing a field in a StarkNet type with a name and type
+#[derive(Serialize, Deserialize, Debug)]
+struct FieldType {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+}
+
+// Structure holding definitions of StarkNet types: StarknetDomain and Challenge
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
is this structure `Deserialize` serialized/deserialized? I can find a piece where it's constructed and eventually destructured, but not {de,}serialized

Reply:
Removed Serialize and Deserialize from here